### PR TITLE
load test app is ready for use

### PR DESCRIPTION
**Pull request by oryband**
please review `cmd/loadtest` and `scripts/report.py`. these are the primary updates in this pr

app is now ready for stress tests, meaning:

- support concurrent tx submitter goroutines
- local sequence number provider, independent of remote horizon
- rate and burst limit
- configuration flags
- proper logs for generating post-test reports from
- also print to multiple logs: stdout and file in json format
- and log colors!
- migrate code from src/ to cmd/benchmark/
  same as other apps in previous commits

### CODE REVIEW

**Comment by oryband** [comment]
please review `cmd/loadtest` and `scripts/report.py`. these are the primary updates in this pr

**Comment by oryband** [comment]
i'll add a readme and better docs in following prs

File: cmd/loadtest/log.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func NewMultiLogger(loggers ...log.Logger) *MultiLogger {
+	return &MultiLogger{
+		loggers: append([]log.Logger{}, loggers...),
+	}
+}
+
+func (l *MultiLogger) Log(keyvals ...interface{}) error {
+	kvs := keyvals[:len(keyvals):len(keyvals)]
```
Comment (bolshoy) [inline cmd/loadtest/log.go:28]: not sure what this index does: `:len(keyvals):len(keyvals)`

**Review by bolshoy — COMMENTED** [review]
👍 
I suggest to reorganise the code and move all the building blocks into `lib` folder, so that the `cmd` folder contains only the specific executables that use the components from lib.

File: cmd/loadtest/log.go (hunk 1 above)
Comment (oryband) [reply cmd/loadtest/log.go:28]: From https://golang.org/ref/spec#Slice_expressions:

> `a[low : high : max]`
constructs a slice of the same type, and with the same length and elements as the simple slice expression a[low : high]. Additionally, it controls the resulting slice's capacity by setting it to `max - low`. Only the first index may be omitted; it defaults to 0. After slicing the array a

**Comment by oryband** [comment]
> I suggest to reorganise the code and move all the building blocks into lib folder, so that the cmd folder contains only the specific executables that use the components from lib.

I did so at first, but I came to the conclusion that every app has slight differences in what should be common code.

Also, the standard for common libraries should be `pkg/` instead of `lib/`: https://peter.bourgon.org/go-best-practices-2016/#repository-structure

**Pull request merged**